WooCommerce/Classes/Authentication/Prologue/LoginOnboardingViewController.swift

```diff
@@ -0,0 +1,142 @@
+import UIKit
+
+/// Contains a feature carousel with buttons that end up on the login prologue screen.
+final class LoginOnboardingViewController: UIViewController {
+    private let stackView: UIStackView = .init()
+    private lazy var pageViewController: LoginProloguePageViewController =
+    LoginProloguePageViewController(pageTypes: [.stats, .orderManagement, .products], showsSubtitle: true)
+    private let onDismiss: () -> Void
+
+    init(onDismiss: @escaping () -> Void) {
+        self.onDismiss = onDismiss
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureMainView()
+        configureCurvedImageView()
+        configureWooLogo()
+        configureStackView()
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return UIDevice.isPad() ? .all : .portrait
+    }
+}
+
+private extension LoginOnboardingViewController {
+    func configureMainView() {
+        view.backgroundColor = .basicBackground
+
+        let bottomView = UIView(frame: .zero)
+        bottomView.backgroundColor = .authPrologueBottomBackgroundColor
+        bottomView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(bottomView)
+        NSLayoutConstraint.activate([
+            bottomView.leadingAnchor.constraint(equalTo: view.safeLeadingAnchor),
+            bottomView.trailingAnchor.constraint(equalTo: view.safeTrailingAnchor),
+            bottomView.heightAnchor.constraint(equalTo: view.heightAnchor, multiplier: 0.3),
+            bottomView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+
+    func configureCurvedImageView() {
+        let imageView = UIImageView(image: .curvedRectangle.withRenderingMode(.alwaysTemplate))
+        imageView.tintColor = .authPrologueBottomBackgroundColor
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(imageView)
+        NSLayoutConstraint.activate([
+            imageView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            imageView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            imageView.topAnchor.constraint(equalTo: view.safeTopAnchor, constant: 52),
+            imageView.heightAnchor.constraint(equalTo: view.heightAnchor, multiplier: 0.7)
```

Comment:
Questions for my learnings:
1. I am trying to understand why you constrained the image height to 0.7 of the view and used the `bottomView` to occupy the remaining 30% of the view. Is the image not scaling well?
2. I am also curious how you got those 0.3 and 0.7 values from the Figma file. I couldn't find anything in Figma representing `bottomView`.

Reply:
Great questions! I didn't get the ratio from Figma, the curved rectangle image view has to be of a certain size and position to match its placement on the existing prologue screen. I didn't go for the exact matching because the prologue screen `WordPressAuthenticator.LoginPrologueViewController` consists of 2 subviews - top prologue UI (from our app `LoginPrologueViewController`) and the button container. In the prologue screen, the curved rectangle view has a top 52 offset and takes the rest of the top container height. In the onboarding screen, the button container in the onboarding view doesn't always match the button container in the prologue screen so I used a 0.7 height ratio after trying a few values.

If we want to match the curved rectangle position 100%, we probably want to update `prologue-curved-rectangle` image to just contain the curved part without the bottom space 😅

Reply:
Thanks for the detailed answer. 😄